src/tester.rs

```diff
@@ -445,35 +463,30 @@ fn run_tests(
         let failures = failures.clone();
         let inner = inner.clone();
         pool.execute(move || {
-            if inner.test_threads == 1 {
+            let single_threaded = inner.test_threads == 1;
+            if single_threaded {
                 eprint!("\ntest lang_test::{} ... ", test_fname);
             }
             let all_str = read_to_string(p.as_path())
                 .unwrap_or_else(|_| fatal(&format!("Couldn't read {}", test_fname)));
             let test_str = inner.test_extract.as_ref().unwrap()(&all_str).unwrap_or_else(|| {
                 fatal(&format!("Couldn't extract test string from {}", test_fname))
             });
+
             if test_str.is_empty() {
-                // Grab a lock on stderr so that we can avoid the possibility of lines blurring
-                // together in confusing ways.
-                let stderr = StandardStream::stderr(ColorChoice::Always);
-                let mut handle = stderr.lock();
-                if inner.test_threads > 1 {
-                    handle
-                        .write_all(&format!("\ntest lang_tests::{} ... ", test_fname).as_bytes())
-                        .ok();
-                }
-                handle
-                    .set_color(ColorSpec::new().set_fg(Some(Color::Yellow)))
-                    .ok();
-                handle.write_all(b"ignored").ok();
-                handle.reset().ok();
-                handle.write_all(b" (test string is empty)").ok();
+                write_ignored(test_fname.as_str(), "test string is empty", single_threaded);
                 num_ignored += 1;
                 return;
             }
 
             let tests = parse_tests(&test_str);
+
+            if !cfg!(unix) && tests.values().find(|t| t.status == Status::Signal).is_some() {
+                write_ignored(test_fname.as_str(), "platform not supported", single_threaded);
```

Comment:
Maybe we can say something like "signal exits not supported on this platform"? It is a slight mouthful, I admit, so if we can be terser, I'm all ears!

Reply:
I can't think of anything better really. Lets go with this.

Reply:
Fixed: 86ff69501c2fb06148d612175836666cdebae040